svn/common.py

```diff
@@ -413,6 +411,25 @@ def diff_summary(self, old, new, rel_path=None):
             diff.append({'path': element.text, 'item': element.attrib['item'], 'kind': element.attrib['kind']})
         return diff
 
+    def diff(self, old, new, rel_path=None):
+        """
+        Provides output of a diff between two revisions (file, change type, file type)
+        """
+        full_url_or_path = self.__url_or_path
+        if rel_path is not None:
+            full_url_or_path += '/' + rel_path
+        diff_result = self.run_command(
+            'diff',
+            ['--old', '{0}@{1}'.format(full_url_or_path, old),
+             '--new', '{0}@{1}'.format(full_url_or_path, new)],
+            combine=True)
+        file_to_diff = {i.split('==')[0].strip(): i.split('==')[-1].strip().strip('=').strip()
```

Comment:
There's a little too much going on in this comprehension and you have to call `i.split('==')` twice.